everestrs/everestrs/src/lib.rs

```diff
@@ -287,3 +340,86 @@ impl Runtime {
         (self.cpp_module).as_ref().unwrap().signal_ready(self);
     }
 }
+
+/// A store for our config values. The type is closely related to
+/// [ffi::ConfigField] and [ffi::ConfigType].
+#[derive(Debug)]
+pub enum Config {
+    Boolean(bool),
+    String(String),
+    Number(f64),
+    Integer(i64),
+}
+
+impl TryFrom<&Config> for bool {
+    type Error = ();
+    fn try_from(value: &Config) -> std::result::Result<Self, Self::Error> {
+        match value {
+            Config::Boolean(value) => Ok(*value),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<&Config> for String {
+    type Error = ();
+    fn try_from(value: &Config) -> std::result::Result<Self, Self::Error> {
+        match value {
+            Config::String(value) => Ok(value.clone()),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<&Config> for f64 {
+    type Error = ();
+    fn try_from(value: &Config) -> std::result::Result<Self, Self::Error> {
+        match value {
+            Config::Number(value) => Ok(*value),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<&Config> for i64 {
+    type Error = ();
+    fn try_from(value: &Config) -> std::result::Result<Self, Self::Error> {
+        match value {
+            Config::Integer(value) => Ok(*value),
+            _ => Err(()),
+        }
+    }
+}
+
+/// Interface for fetching the configurations through the C++ runtime.
+pub fn get_module_configs() -> HashMap<String, HashMap<String, Config>> {
+    let args: Args = argh::from_env();
+    let raw_config = ffi::get_module_configs(
+        &args.module,
+        &args.prefix.to_string_lossy(),
+        &args.conf.to_string_lossy(),
+    );
+
+    // Map the nested Vec's into a nested HashMaps.
+    raw_config
+        .into_iter()
+        .map(|ffi::RsModuleConfig { module_name, data }| {
+            let current = data
+                .into_iter()
+                .map(|field| {
+                    let value = match field.config_type {
+                        ffi::ConfigType::Boolean => Config::Boolean(field.bool_value),
+                        ffi::ConfigType::String => Config::String(field.string_value),
+                        ffi::ConfigType::Number => Config::Number(field.number_value),
+                        ffi::ConfigType::Integer => Config::Integer(field.integer_value),
+                        _ => panic!("This should never happen"),
```

Comment:
not a helpful error message. Please include the unexpected value. Or use `unreachable!`.

Reply:
It should never happen however if c++ casts an invalid value to enum class it should be UB and I'm not sure what then happens here - will print the value